processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -1173,6 +1174,28 @@ else if ( mapping.getJavaExpression() != null ) {
                     }
                 }
             }
+
+            //check if there's a condition and if the method exists
+            if ( mapping.getCondition() != null ) {
```

Comment:
Tried to do this check in a few different places but this one seems to be the most appropriate one. Could somebody please confirm? If so, I'll add a check for the conditional method param type.